src/or/proto_socks.c

```diff
@@ -338,6 +338,101 @@ process_socks5_methods_request(socks_request_t *req, int have_user_pass,
   return res;
 }
 
+static int
+parse_socks5_userpass_auth(const uint8_t *raw_data, socks_request_t *req,
+                           size_t datalen, size_t *drain_out)
+{
+  int res = 1;
+  socks5_client_userpass_auth_t *trunnel_req = NULL;
+  ssize_t parsed = socks5_client_userpass_auth_parse(&trunnel_req, raw_data,
+                                                     datalen);
+  
+  if (parsed == -1) {
+    log_warn(LD_APP, "socks5: parsing failed - invalid user/pass "
+                     "authentication message.");
+    res = -1;
+    goto end;
+  } else if (parsed == -2) {
+    res = 0;
+    goto end;
+  }
+
+  *drain_out = (size_t)parsed;
```

Comment:
Let's add an assertion, and let's initialize drain_out to 0.

Reply:
Fixed.